src/renderer/components/Visualiser/VisualiserContent.vue

```diff
@@ -134,34 +133,31 @@ export default {
       this.setActiveTab(`tab-${tab}`);
     },
     newTab() {
-      const newTabId = Math.max(...Array.from(this.tabs.keys())) + 1; // Get max tab id and increment it for new tab id
-      this.tabs.set(newTabId, undefined);
+      const newTabId = `${Math.max(...Object.keys(this.tabs)) + 1}`; // Get max tab id and increment it for new tab id
+      this.tabs[newTabId] = undefined;
       this.currentTab = newTabId;
       this.setActiveTab(`tab-${newTabId}`);
     },
     async closeTab(tab) {
-      // Find tab compoenent which has same tabId as tab to be closed and destroy it manually
-      const visTab = this.$children.find(x => (x.tabId && x.tabId === tab));
-      visTab.$destroy();
-
-      this.tabs.delete(tab);
-
-      // if tab being closes is same as current tab switch to first tab in tabs
-      if (this.currentTab === tab) this.currentTab = Array.from(this.tabs.keys())[0];
-      else { // re-set the same tab to trigger dynamic rerendering of list
-        const temp = this.currentTab;
-        this.currentTab = null;
-        this.currentTab = temp;
-      }
+      const visTabToClose = this.$children.find(x => x.tabId && x.tabId === tab);
+      visTabToClose.$destroy();
+
+      // vue doesn't detect property deleteion, so we clone, delete and reassign
+      const tabsClone = { ...this.tabs };
+      delete tabsClone[tab];
```

Comment:
this doesn't end up re-drawing the tab does it?

Reply:
no, it wouldn't. to understand why this cloning is required, one would have to understand the Vue reactivity in-depth. that's why I've left a comment on line 145 to explain the reason for cloning.

Reply:
fair enough!